include/solvers/eigen_sparse_linear_solver.h

```diff
@@ -75,8 +75,8 @@ class EigenSparseLinearSolver : public LinearSolver<T>
   solve (SparseMatrix<T> & matrix,
          NumericVector<T> & solution,
          NumericVector<T> & rhs,
-         const double tol,
-         const unsigned int m_its) override;
+         const std::optional<double> tol = std::nullopt,
```

Comment:
I always struggled to see a point to `std::optional<T>`, vs just using a `T*` and interpreting `nullptr` as the null option ... but this sort of backwards API compatibility is really cool.